### PR TITLE
UCS/ASYNC: Add API to change fd events mask.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3934/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich @shamisp pls review

File: test/gtest/ucs/test_async.cc [hunk 1]
```diff
@@ -494,6 +494,29 @@ UCS_TEST_P(test_async, ctx_timer_block) {
...
+    local_event le(GetParam());
+
+    le.push_event();
+    suspend_and_poll(&le, COUNT);
+    EXPECT_GE(le.count(), 1);
+
+    ucs_async_modify_handler(le.event_id(), 0);
+    le.push_event();
+    suspend_and_poll(&le, COUNT);
+    EXPECT_GE(le.count(), 0);
```
Comment (brminich) [inline test/gtest/ucs/test_async.cc]: I guess you want to check that le.count did not change. Seems that you need to compare for equality with previous counter value

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3942/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
bot:ornl:retest

**Pull request merged**